File: src/NetWorthCalculator.Web.ReactJs/src/Services/BalanceSheet/Dto/BalanceSheetDto.ts
import { list, object, serializable } from "serializr";
import { Currency } from "../../../Models/Currency/Currency";
import AssetDto from "./AssetDto";
import LiabilityDto from "./LiabilityDto";

export default class BalanceSheetDto {
	@serializable id!: number;

	@serializable currency!: Currency;

	@serializable netWorth!: number;

	@serializable totalAssets!: number;

	@serializable totalLiabilities!: number;

	@serializable(list(object(AssetDto))) assets!: AssetDto[];

	@serializable(list(object(AssetDto))) liabilities!: LiabilityDto[];
}

File: src/NetWorthCalculator.Web.Service/Controllers/BalanceSheet/Dto/BalanceSheetDto.cs
using NetWorthCalculator.Entities.Enums;
using System.Collections.Generic;

namespace NetWorthCalculator.Web.Service.Controllers.BalanceSheet.Dto
{
	public class BalanceSheetDto
	{
		public int Id { get; set; }

		public Currency Currency { get; set; }

		public decimal NetWorth { get; set; }

		public decimal TotalAssets { get; set; }

		public decimal TotalLiabilities { get; set; }

		public List<AssetDto> Assets { get; set; }

		public List<LiabilityDto> Liabilities { get; set; }
	}
}
